make/compiler/Makefile.gnu

```diff
@@ -405,14 +405,14 @@ RUNTIME_CXXFLAGS += $(WARN_CXXFLAGS)
 #
 # When compiling GASNet...
 #
-# -Wno-missing-prototypes must be used because
+# -Wno-missing-prototypes must be used becausex
 # gasneti_linkconfig_idiotcheck isn't prototyped before it's used (boo)
 # and is static, so we can't prototype it for them (awwww).
 #
 # -Wno-strict-prototypes has to be used because most GASNet prototypes
 # aren't strict.
 #
-CHPL_GASNET_MORE_CFLAGS = -Wno-strict-prototypes -Wno-missing-prototypes
+CHPL_GASNET_MORE_CFLAGS = -Wno-strict-prototypes -Wno-missing-prototypes -Wno-cast-function-type-strict
```

Comment:
Do older gccs that we might need to compile Chapel with support this flag `-Wno-cast-function-type-strict`? I would imagine that they don't. If not, we need to check the gcc version before adding this flag. There are other examples of this in this file.

Reply:
I don't believe gcc supports the flag, I think this can be removed. I will do fresh builds of this PR to double check